pkg/server/commands/read_changes.go

```diff
@@ -15,10 +17,11 @@ import (
 )
 
 type ReadChangesQuery struct {
-	backend       storage.ChangelogBackend
-	logger        logger.Logger
-	encoder       encoder.Encoder
-	horizonOffset time.Duration
+	backend         storage.ChangelogBackend
+	logger          logger.Logger
+	encoder         encoder.Encoder
```

Comment:
Should we rename this to `encoder.TokenEncoder`? :)

Reply:
Right...tbh I'm thinking if we can remove the encoder and merge it with the serializer. that might be a bit more effort though